### PR TITLE
Add new course settings to update course flag Destination -> Source

### PR DESCRIPTION
**Pull request by tehreem-sadat**
Add new course settings for destination course -> update course flag from Destination to Source.
Styling is pending to convert this checkbox in Switch.

image
image

### CODE REVIEW

File: openedx/features/wikimedia_features/meta_translations/utils.py [hunk 1]
```diff
@@ -300,3 +300,18 @@ def get_block_status(block_id):
...
+def update_course_to_source(course_key):
+    try:
+        translation_link = CourseTranslation.objects.get(course_id=course_key)
+        log.info('Start converting course with id: {} to Source'.format(str(course_key)))
+        # update all underlying component's flag to source
+        course = get_course_by_id(course_key)
+        log.info('Check and update all underlying blocks to Source'.format(str(course_key)))
+        for course_block in CourseBlock.objects.filter(course_id=course_key):
+            course_block.update_flag_to_source(course.language)
+            log.info('Update course block with id: {} to Source'.format(str(course_key)))
```
[the remark this answers is not in the document]
Comment (tehreem-sadat) [reply openedx/features/wikimedia_features/meta_translations/utils.py:313]: If the user is updating the course flag then there are more possible chances that he wants to create a new versioned course from it then in that case, we need to create these blocks again. 

Let's see how things are working so maybe we'll consider this requirement in the future.

File: cms/templates/settings.html [hunk 2]
```diff
@@ -320,6 +320,21 @@ <h2 class="title-2">${_('Course Schedule')}</h2>
...
+        <div class="group-settings course_flag">
+          <header>
+            <h2 class="title-2">${_('Course Direction Flag')}</h2>
+            <span class="tip">${_('Convert course to Source')}</span>
+          </header>
+          <div>
+              % if is_destination_course:
+                <input type="checkbox" id="is-destination-course">
+              % else:
+                <input type="checkbox" id="is-destination-course" disabled>
```
[the remark this answers is not in the document]
Comment (tehreem-sadat) [reply cms/templates/settings.html:332]: great catch! I should add that as well.

File: openedx/features/wikimedia_features/meta_translations/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ali-Salman29) [reply openedx/features/wikimedia_features/meta_translations/utils.py:313]: Oky,

File: cms/templates/settings.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (tehreem-sadat) [reply cms/templates/settings.html:332]: done.
image

**Review by Ali-Salman29 — APPROVED** [review]

**Pull request merged**